server/controllers/payment.js

```diff
@@ -0,0 +1,80 @@
+require("dotenv").config();
+const PUBLIC_STRIPE_API_KEY = process.env.PUBLIC_STRIPE_API_KEY;
+const SECRET_STRIPE_API_KEY = process.env.SECRET_STRIPE_API_KEY;
+const stripe = require("stripe")(SECRET_STRIPE_API_KEY); // ! remove API_KEY before commit
+const Payment = require("../models/payment");
+const itemLookup = require("../helper/itemLookup");
+const https = require("https");
+
+const handleError = (e, res) =>
+  e.status && e.message
+    ? res.status(e.status).send(e.message)
+    : res.status(400).send(e);
+
+const getKey = (req, res, next) => {
+  try {
+    return res.status(200).send({ STRIPE_API_KEY: PUBLIC_STRIPE_API_KEY });
```

Comment:
Any reason why you want to get the key instead of just storing it on the front-end (since it is a public stripe key) as a react environment variable? I'm assuming this might be better practice, just want to confirm for my own learning actually

Reply:
I had no best practice thoughts about this, other than avoiding hard coding anything on the FE. I guess my thought was that it would be more scalable if different transactions could be routed through different Stripe accounts with different keys.

But looking back at the docs, I think this way of doing things triggers unnecessary re-instantiation of the Stripe object on each render. [Add Stripe.js to your page.](pk_test_51H9aiQHBjJBD671jjlNd2YSaGTgUaRtAfrSOlF1271z4ftGZwKSP2UVI0XWHuQqzA5NAY7pv5FvuQj1LT47nokHG00DZWM1eoW) This shouldn't be an issue, since we should probably lock down the Payment page and keep any re-renders within the CheckoutForm, but feels a little iffy.

Reply:
Okay, the only way to do this properly would be to lift the stripe object out of this component, I hardcoded it in my next commit.